dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/InterpretationController.java

```diff
@@ -75,6 +89,56 @@
     @Autowired
     private IdentifiableObjectManager idObjectManager;
 
+    @Override
+    @SuppressWarnings( "unchecked" )
+    protected List<Interpretation> getEntityList( WebMetadata metadata, WebOptions options, List<String> filters,
+        List<Order> orders )
+        throws QueryParserException
+    {
+
+        // If mentions:in:[username] in filters -> Remove from filters and add
+        // as disjunction
+        List<String> mentions = new ArrayList<String>();
+        ListIterator<String> filterIterator = filters.listIterator();
+        while ( filterIterator.hasNext() )
+        {
+            String[] filterSplit = filterIterator.next().split( ":" );
+            if ( filterSplit[1].equals( "in" ) && filterSplit[0].equals( "mentions" ) )
+            {
+                mentions.add( filterSplit[2] );
+                filterIterator.remove();
+            }
+        }
```

Comment:
101-110: This is too much logic for a controller, can we move to InterpretationService or to MentionUtils?

Reply:
Done here https://github.com/dhis2/dhis2-core/pull/1453/commits/a1b623680b81399762e4e0aa70230ce1827e911f 
I thought about this approach before but IMHO it makes the code a bit more unreadable. Also as we don't have access to `org.hisp.dhis.query` from dhis-api, therefore, I had to return a `Collection<?>` which is not particularly elegant.